### ISSUE TITLE
problem with 'ldap filter' setting

### ISSUE THREAD

**Issue by silas101** [jonmbake/discourse-ldap-auth#13]
In LDAP settings, for the "LDAP Filter" field, I'm using the following syntax:

memberOf=CN=Group,OU=Users,DC=Domain,DC=com

Authentication works for the user if that user is the only member of "Group". However, if more than one member is added to "Group", only the first user account on the group membership list is able to authenticate.

**Comment by jonmbake** [comment]
Hey @silas101. This plugin simply delegates to https://github.com/intridea/omniauth-ldap to perform the LDAP authentication... If there is a problem with the filter, it could be a bug in that project.

**Comment by kierandelaney** [comment]
Hi - I know it's been a long time but this is exactly the problem we encountered and it's to do with the query going to LDAP, not a bug in either the plugin or omniauth.

Doing `memberOf=CN=Group,OU=Users,DC=Domain,DC=com` isn't enough, you also need to include in your filter the search element that restricts to the user attempting to login, eg 

`&(sAMAccountName=%{username})(memberOf=CN=Group,OU=Users,DC=Domain,DC=com)`

Obviously you'll need to use whatever theLDAP attribute name for the user name is in place of sAMAccountName (Windows AD).

The above also failed - we had to wrap the entire thing in brackets, not something we were used to when querying from PHP.
